### DISCUSSION TITLE
Inject PlantUml Diagrams to Markdown/AsciDoc

### DISCUSSION THREAD

**Discussion by dawidci-papaya** [avisi-cloud/structurizr-site-generatr#410] [Q&A] [answered]
Hi,
I saw possibility to inject  mermaid diagrams into Markdown/AsciDoc,  but i would like to ask if it is possible to inject diagrams in plantuml code directly to description files ?

I use https://github.com/avisi-cloud/structurizr-site-generatr/discussions/351  to generate sequence diagrams from C4 Dynamic diagrams. It works good but for some complex use cases (Conditions in diagrams) i would like to generate my own diagram and inject it as plant uml code.

**Comment by qtzar** [comment]
I currently have a PR open for Image Views. While I've only tested with actual images ( jpg/png ) the image view capability does allow for image generation from plantulm files if you define an image generator. The element identifier will make the image show up under the correct software system in the generated site but if you define the key then you may be able to make the image show in a markdown document with the embed keyword like you can use for landscapes or other diagrams. Would have to be tested.

```
image <*|element identifier> [key] {
    plantuml <file|url>
}

views {
    properties {
        "plantuml.url" "http://localhost:7777"
        "plantuml.format" "svg"
   }
```

`![Image Diagram](embed:My-Image-View-Key)`

[Structurizr DSL Language Reference - Image Views](https://docs.structurizr.com/dsl/language#image-view)

**Reply by dirkgroot** [reply]
The PR you have open actually has very limited support for image views. It just provides support for image views containing an `image`. This is fine, of course, but  @dawidci-papaya's question is about having the generator render embedded PlantUML code in a Markdown file, similar to how we support rendering embedded mermaid diagrams. For PlantUML this would look like this:

````markdown
# Example

Here's an embedded PlantUML diagram:

```puml
@startuml

Foo -> Bar: doSomething()

@enduml
```
````

This is currently not supported, but it surely is possible to add this functionality. I also think this would be a useful addition, especially since we already support this for Mermaid diagrams.

I'll look into it.

**Comment by dirkgroot** [comment]
I've created a draft PR (#411), which currently adds support for rendering PlantUML diagrams inside Markdown files. I'm still trying to figure out how to do the same for Asciidoc.

**Comment by dirkgroot** [comment, marked as answer]
Since PR #411 has been merged, and this functionality is available in the current release, this is now officially done 😄.
